qcore/geo.py

```diff
@@ -149,6 +149,60 @@ def ll2gp_multi(
     return xy
 
 
+def oriented_bearing_wrt_normal(
+    from_direction: np.ndarray, to_direction: np.ndarray, normal: np.ndarray
+) -> float:
+    """Compute the oriented bearing between two directions with respect to a normal.
+
+    This function is useful to calculate, for example, strike and dip
+    directions. The orientation is established via the right hand rule
+    (or refer to the diagram below).
+
+        to_bearing
+           ^
+           │
+           │
+           │
+           │
+           │
+           │<┐  bearing
+           │ └─┐
+           ╳─────────────> from_direction
+          ╱
+         ╱
+        ╱
+       ╱
+    normal
+
+    Parameters
+    ----------
+    from_direction : np.ndarray
+        The direction to measure the bearing from.
+    to_direction : np.ndarray
+        The direction to measure the bearing to.
+    normal : np.ndarray
+        The normal direction to orient the bearing with via the right hand rule.
+
+    Returns
+    -------
+    float
+        The bearing from u to v with oriented with respect to normal.
```

Comment:
I think the first "with" in "The bearing from u to v with oriented with respect to normal." should be deleted

Reply:
yeah good catch.